chain/bitcoind_client.go

```diff
@@ -469,6 +469,102 @@ func (c *BitcoindClient) RescanBlocks(
 	return rescannedBlocks, nil
 }
 
+// RescanBlocksBatched rescans any blocks passed in batched manner using asyn
+// calls, returning only the blocks that matched as []btcjson.BlockDetails.
+func (c *BitcoindClient) RescanBlocksBatched(
+	blockHashes []chainhash.Hash) ([]btcjson.RescannedBlock, error) {
+
+	// requests slice will store rpcclient.FutureGetBlockHeaderVerboseResult
+	// for each blockHash index.
+	requests := make(
+		[]rpcclient.FutureGetBlockHeaderVerboseResult, 0, len(blockHashes),
+	)
+
+	for i := range blockHashes {
+		hash := blockHashes[i]
+		headerRequest, err := c.GetBlockHeaderVerboseAsync(&hash)
+		if err != nil {
+			log.Warnf(
```

Comment:
nit: `Warnf` can use the compact formatting as it's only logging, so can look the same as L506.